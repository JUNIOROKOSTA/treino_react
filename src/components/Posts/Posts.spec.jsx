import { render, screen } from '@testing-library/react';
import { Posts } from '.';

const props = {
  posts: [
    {
      title: 'Title 1',
      body: 'Body 1',
      id: 1,
      cover: 'img/img1.png',
    },
    {
      title: 'Title 2',
      body: 'Body 2',
      id: 2,
      cover: 'img/img2.png',
    },
    {
      title: 'Title 3',
      body: 'Body 3',
      id: 3,
      cover: 'img/img3.png',
    },
  ],
};

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts {...props} />);

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);

    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);

    expect(screen.getAllByText(/body/i)).toHaveLength(3);

    expect(screen.getByRole('img', { name: /title 1/i })).toHaveAttribute('src', 'img/img1.png');
  });

  // Create snapshot of test

  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
