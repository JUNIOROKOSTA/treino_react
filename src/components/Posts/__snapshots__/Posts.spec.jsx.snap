// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title 1
      </h1>
      <h3>
        Body 1
      </h3>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title 2
      </h1>
      <h3>
        Body 2
      </h3>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title 3
      </h1>
      <h3>
        Body 3
      </h3>
    </div>
  </div>
</div>
`;
